graph_train.py

```diff
@@ -1,183 +0,0 @@
-        #     print()
-        return loss, accuracy
-
-    def configure_optimizers(self):
-        self.optimizer = torch.optim.Adam(self.parameters(), lr=0.01)
-        return self.optimizer
-
-    def training_step(self, batch, batch_idx):
-        loss, acc = self.forward(batch, mode='train')
-        self.log("train_loss", loss, on_step=False, on_epoch=True)
-        self.log("train_acc", acc, on_step=False, on_epoch=True)
-        return loss
-
-    def validation_step(self, batch, batch_idx):
-        loss, acc = self.forward(batch, mode='val')
-        self.log("val_loss", loss)
-        self.log("val_acc", acc)
-
-    def test_step(self, batch, batch_idx):
-        loss, acc = self.forward(batch, mode='test')
-        self.log("test_loss", loss)
-        self.log("test_acc", acc)
-
-
-class GenerateCallback(pl.Callback):
-    def __init__(self):
-        """
-        Args:
-            gamma (float, optional): [description]. Defaults to 0.2.
-        """
-        super().__init__()
-
-    def on_epoch_end(self, trainer, pl_module):
-        """
-        This function is called after every epoch.
-        In this function, the dev accuracy is checked and the learning rate is adjusted if dev accuracy has decreased
-        """
-        val_acc = trainer.callback_metrics.get('val_acc').item()
-        print('Epoch [' + str(trainer.current_epoch + 1) + ']; train_loss: ' + str(
-            trainer.callback_metrics.get('train_loss').item())
-              + '; val_loss: ' + str(trainer.callback_metrics.get('val_loss').item()) + ': val_acc: ' + str(val_acc))
-
-
-def train_model(args):
-    os.makedirs(args.log_dir, exist_ok=True)
-
-    if args.dataset == 'r8':
-        dataset = R8Graph(device)
-    elif args.dataset == 'r52':
-        dataset = R52Graph(device)
-
-    # data = data_object.data.to(device)
-    graph_data_loader = geom_data.DataLoader(dataset, batch_size=1)
-
-    # Create a PyTorch Lightning trainer with the generation callback
-    gen_callback = GenerateCallback()
-    trainer = pl.Trainer(default_root_dir=args.log_dir,
-                         checkpoint_callback=ModelCheckpoint(save_weights_only=True, mode="min", monitor="val_loss"),
-                         gpus=1 if torch.cuda.is_available() else 0,
-                         max_epochs=args.epochs,
-                         callbacks=[gen_callback],
-                         progress_bar_refresh_rate=1 if args.progress_bar else 0)
-    trainer.logger._default_hp_metric = None
-
-    # Create model
-    pl.seed_everything(args.seed)
-    model = Graph_model(len(dataset.iton))
-
-    # optionally resume from a checkpoint
-    if args.resume != None:
-        if os.path.isfile(args.resume):
-            print('=> loading checkpoint %s' % args.resume)
-            model = NLI_model.load_from_checkpoint(args.resume)
-            print('=> loaded checkpoint %s' % args.resume)
-        else:
-            print('=> no checkpoint found at %s' % args.resume)
-
```

Comment:
It would be nice to have these line above to be able to continue from a checkpoint also in train.py